RELEASE.md

```diff
@@ -0,0 +1,57 @@
+# Releasing Stark
+
+## Pre-reqs
+### Local
+On your local machine, you must configure the `GITHUB_TOKEN` environment variable.
+It will be used by release-it to push to and create the release page on GitHub (cfr release:prepare section below).
+
+### Travis
+On Travis, the following should be configured:
+* NPM_TOKEN environment variable
+  * if 2FA is enabled for the account the only auth-only level can be used: https://docs.npmjs.com/getting-started/using-two-factor-authentication#levels-of-authentication
+  * that variable MUST NEVER be logged/exposed. If exposed then the token MUST be revoked and the account password changed ASAP
+
+## Changelog
+First of all: *Never* edit CHANGELOG.md manually!
+
+The changelog will be updated automatically as part of the release process and based on the commit log using conventional-changelog (https://github.com/conventional-changelog)
+We use the Angular format for our changelog and for it to work properly, please make sure to respect our commit conventions (see CONTRIBUTING guide).
+
+## Creating a release
+Make sure that:
+* all changes have merged into master
+* everything is up to date locally
+* everything is clean locally
+* execute `npm run release`
+
+Enjoy the show.
+
+## Publishing the release on npm
+Once you have pushed the tag, Travis will handle things from there
+
+## What happens once a release is triggered
+
+### release
+* first we make sure that there are no local changes (if there are we stop right there)
+* then we execute release-it: https://github.com/webpro/release-it which
+  * bumps the version in the root package.json automatically (determines the bump type to use depending on the commit message logs)
+    * that version number will be used as basis in the build to adapt all other package.json files
+  * generates/updates the CHANGELOG.md file using: conventional-changelog: https://github.com/conventional-changelog
+  * commits both package.json and CHANGELOG.md
+  * creates a new git tag and pushes it
+  * creates a github release page and makes it final
+
+After this, the release is tagged and visible on github
+
+### publish
+Once the tag is pushed to GitHub, Travis picks it up and initiates a build.
+Travis executes builds, tests, then executes `npm run release:publish`.
+
+That script makes some checks then, if all succeed it publishes the different packages on npm.
```

Comment:
Question... what if something goes wrong during these checks? what if this process fails? how to re-trigger the publish?

Reply:
Let's be optimistic for now: we'll use tag until we face an issue.
If we do face an issue, then we can switch to checking the commit message instead and make sure we can do the publish locally and not only through Travis